sycl/include/sycl/kernel_bundle.hpp

```diff
@@ -26,12 +26,15 @@
 #include <sycl/ext/oneapi/properties/property.hpp>       // build_options
 #include <sycl/ext/oneapi/properties/property_value.hpp> // and log
 
-#include <array>       // for array
-#include <cstddef>     // for std::byte
-#include <cstring>     // for size_t, memcpy
-#include <functional>  // for function
-#include <iterator>    // for distance
-#include <memory>      // for shared_ptr, operator==, hash
+#include <array>      // for array
+#include <cstddef>    // for std::byte
+#include <cstring>    // for size_t, memcpy
+#include <functional> // for function
+#include <iterator>   // for distance
+#include <memory>     // for shared_ptr, operator==, hash
+#ifdef __cpp_lib_span
+#include <span> // for span
+#endif
```

Comment:
The [standard](https://eel.is/c++draft/version.syn#:macro,feature-test) says that `__cpp_lib_span` will only be defined if:
- You include `<version>`; or
- You include `<span>`.

So, I think this should use [`__has_include`](https://eel.is/c++draft/cpp.cond#:__has_include) instead:

```suggestion
#if __has_include(<span>)
#include <span> // for span
#endif
```

The later uses of `__cpp_lib_span` are correct, because they actually guard against _use_ of `std::span`.